resources/helm/dask-gateway/values.yaml

```diff
@@ -54,6 +54,23 @@ gateway:
       # Configuration fields to set on the authenticator class.
       options: {}
 
+  livenessProbe:
+    # Enables the livenessProbe. 
+    enabled: true
+    # Configures the livenessProbe. 
+    initialDelaySeconds: 5
+    timeoutSeconds: 2
+    periodSeconds: 10
+    failureThreshold: 3
+  readinessProbe:
+    # Enables the readinessProbe.
+    enabled: true
+    # Configures the readinessProbe.
+    initialDelaySeconds: 5
+    timeoutSeconds: 2
+    periodSeconds: 10
+    failureThreshold: 1
```

Comment:
Are these good defaults? I think the period and timeout seem fine, but am a bit concerned about how small the `failureThreshold` is for readiness. The default is 3, is there a good reason to use 1 here? The `initialDelaySeconds` also might be a bit small for liveness, if the process is slow to start it only has 35 seconds before it will be killed and restarted - should we worry about that?

I do agree that the default time ratio between failing readiness and liveness should be 3, that seems reasonable. Thoughts @droctothorpe?

Reply:
I'm happy to substitute with other values. My research into sensible defaults wasn't exhaustive. I also consulted with a member of our SRE team before setting them.

Here's some of the reasoning:
* `initialDelaySeconds` -- AFAIK, this doesn't start until the container is already up and running. Since everything is preconfigured, the actual process start time is really fast. I even tested a 3 second `initialDelaySeconds` and ran into no issues. 
* livenessProbe `failureThreshold` is pretty low because if the API can't respond to a health check, it probably can't respond to other kinds of traffic. `2` might be a reasonable alternative if you want it to keep sending traffic for another 10 seconds. Also, if you have a second API deployed (which is advisable in a production environment), you'd want to start routing traffic to the healthy pod rather than the unhealthy one sooner rather than later. 
* `failureThreshold` is set to 3 (with a `periodSeconds` of 10 for a total of 30 seconds of failed health checks) because, by that point, the API has been unresponsive for 30 seconds and it only takes a few seconds to reap and replace the pod. Fundamentally, the pods are disposable and can be rapidly replaced. 

Totally open to alternatives.

Reply:
I think this makes sense for a production deployment (with `replicas > 1`). Since the default is 1 replica, and many users won't change that, I think we want to be less aggressive by default.

Perhaps `failureThreshold` of `3` for `readinessProbe`, and `6` for `livenessProbe`? That keeps the time delay between failing readiness and liveness, but also adds some slack for users running the chart unmodified.

Reply:
Copy that. Pushed the update.